packages/react-core/src/hooks/useUserAttributes/__tests__/useUserAttributes.test.ts

```diff
@@ -0,0 +1,379 @@
+import { act, renderHook } from '@testing-library/react-hooks';
+
+import {
+  SendUserAttributeVerificationCodeOutput,
+  UpdateUserAttributesOutput,
+  UserAttributeKey,
+  confirmUserAttribute,
+  deleteUserAttributes,
+  fetchUserAttributes,
+  sendUserAttributeVerificationCode,
+  updateUserAttributes,
+} from '@aws-amplify/auth';
+
+import * as AuthModule from '@aws-amplify/auth';
+import {
+  ConfirmInput,
+  DeleteInput,
+  FetchInput,
+  HandleAttributeActionOutput,
+  SendCodeInput,
+  UpdateInput,
+  VerifiableAttribute,
+} from '../types';
+import useUserAttributes from '../useUserAttributes';
+
+const confirmUserAttributeSpy = jest.spyOn(AuthModule, 'confirmUserAttribute');
+const deleteUserAttributesSpy = jest.spyOn(AuthModule, 'deleteUserAttributes');
+const fetchUserAttributesSpy = jest.spyOn(AuthModule, 'fetchUserAttributes');
+const sendCodeSpy = jest.spyOn(AuthModule, 'sendUserAttributeVerificationCode');
+const updateUserAttributesSpy = jest.spyOn(AuthModule, 'updateUserAttributes');
+
+const mockAttributes: { [Attribute in UserAttributeKey]?: string } = {
+  email: 'test@mail.com',
+  name: 'Name',
+  nickname: 'Nickname',
+  locale: 'Testville',
+};
+
+const mockDeletion: { [Attribute in UserAttributeKey]?: string } = {
+  email: 'test@mail.com',
+  name: 'Name',
+};
+
+const mockError = new Error('TESTING ERROR');
+
+const mockConfirmInput: ConfirmInput = {
+  type: 'CONFIRM',
+  userAttributeKey: 'email',
+  confirmationCode: '000000',
+};
+
+const mockAfterVerification: HandleAttributeActionOutput = {
+  attributes: { email: 'test@mail.com', name: 'name' },
+  pendingVerification: [{ name: 'email' }],
+};
+
+const mockDeleteInput: DeleteInput = {
+  type: 'DELETE',
+  userAttributeKeys: ['locale', 'nickname'],
+};
+
+const mockFetchInput: FetchInput = { type: 'FETCH' };
+
+const mockUpdateInput: UpdateInput = {
+  type: 'UPDATE',
+  userAttributes: {
+    name: 'New Name',
+    nickname: 'New Nickname',
+    email: 'new@mail.com',
+  },
+};
+
+const mockUpdateOutput: UpdateUserAttributesOutput = {
```

Comment:
This part is weird. The API function updateUserAttributes returns the same shape object, but only with the attributes submitted to the update. However, the JS-specified return type MUST include every standard attribute.